simbad/util/__init__.py

```diff
@@ -126,3 +129,46 @@ def tmp_file(delete=False, directory=None, prefix="tmp", stem=None, suffix=""):
         if not delete:
             open(tmpf, 'w').close()
         return tmpf
+
+
+def submit_chunk(collector, run_dir, nproc, job_name, submit_qtype, submit_queue, monitor, success_func):
+    """Submit jobs in small chunks to avoid using too much disk space
+
+    Parameters
+    ----------
+    collector : list
+        :obj:`~pyjob.script.ScriptCollector` containing run scripts
+    nproc : int, optional
+        The number of processors to run the job on
+    job_name : str
+        The name of the job to submit
+    submit_qtype : str
+        The cluster submission queue type - currently support SGE and LSF
+    submit_queue : str
+        The queue to submit to on the cluster
+    success_func : func
+        function to check for success
+
+    """
+
+    if submit_qtype == 'local':
+        processes = nproc
+        array_size = None
+    else:
+        processes = None
+        array_size = nproc
+
+    with TaskFactory(submit_qtype,
+                     collector,
+                     cwd=run_dir,
+                     name=job_name,
+                     processes=processes,
+                     max_array_size=array_size,
+                     queue=submit_queue,
+                     permit_nonzero=True,
```

Comment:
This is dangerous and you may not always want it. You basically allow a non-zero return code from any script run. My suggestion is to disable it by default and require explicit setting from calling this function